dashboard/service.go

```diff
@@ -0,0 +1,128 @@
+package dashboard
+
+import (
+	"github.com/gin-contrib/sessions"
+	"github.com/gin-contrib/sessions/cookie"
+	"github.com/gin-gonic/gin"
+	"github.com/nknorg/nkn/dashboard/auth"
+	serviceConfig "github.com/nknorg/nkn/dashboard/config"
+	"github.com/nknorg/nkn/dashboard/routes"
+	"github.com/nknorg/nkn/node"
+	"github.com/nknorg/nkn/util/config"
+	"github.com/nknorg/nkn/util/log"
+	"github.com/nknorg/nkn/vault"
+	"github.com/pborman/uuid"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+var (
+	localNode *node.LocalNode
+	wallet    vault.Wallet
+)
+
+func Init(ln *node.LocalNode, w vault.Wallet) {
+	serviceConfig.IsInit = true
+	localNode = ln
+	wallet = w
+}
+
+func Start() {
+	// build release settings
+	gin.SetMode(gin.ReleaseMode)
+	app := gin.New()
+	app.Use(gin.LoggerWithFormatter(func(param gin.LogFormatterParams) string {
+		log.WebLog.Infof("%s - [%s] \"%s %s %s %d %s\" %s \"%s\"",
+			param.ClientIP,
+			param.TimeStamp.Format(time.RFC1123),
+			param.Method,
+			param.Path,
+			param.Request.Proto,
+			param.StatusCode,
+			param.Latency,
+			param.Request.UserAgent(),
+			param.ErrorMessage,
+		)
+		return ""
+	}))
+
+	app.Use(gin.Recovery())
+
+	store := cookie.NewStore([]byte("nkn"))
```

Comment:
It seems we should pass a secret key instead of a constant to NewStore: https://godoc.org/github.com/gin-contrib/sessions/cookie
I haven't used this before so I may be wrong about it, but we should investigate more on this one since it's security related.